File: test backend/backend/app/routers/gpu_products.py
from fastapi import APIRouter, HTTPException
from app.services.gpu_products_service import list_grouped_gpus , get_gpu_details
from pydantic import BaseModel
from typing import Optional, Dict

router = APIRouter(prefix="/api/products/gpu", tags=["GPU Products"])

class GpuListRequest(BaseModel):
    search: Optional[str] = None
    sort: str = "price_asc"
    limit: int = 24
    offset: int = 0
    filters: Dict = {}

@router.post("")
def post_gpu_products(req: GpuListRequest):
    return list_grouped_gpus(
        req.search,
        req.filters,
        req.sort,
        req.limit,
        req.offset,
    )

@router.get("/{model}/{vram}")
def get_gpu_product_details(model: str, vram: int):
    data = get_gpu_details(model, vram)
    if not data:
        raise HTTPException(404, "GPU not found")
    return data

File: frontend/src/api/gpuProducts.js
import { apiFetch } from "./api";

export function fetchGpus({
  search = null,
  sort = "price_asc",
  limit = 24,
  offset = 0,
  filters = {},
} = {}) {
  const params = new URLSearchParams();

  if (search) params.append("search", search);
  params.append("sort", sort);
  params.append("limit", limit);
  params.append("offset", offset);

  if (filters.brand)     params.append("brand",     filters.brand);
  if (filters.min_vram)  params.append("min_vram",  filters.min_vram);
  if (filters.min_price) params.append("min_price", filters.min_price);
  if (filters.max_price) params.append("max_price", filters.max_price);

  return apiFetch(`/api/products/gpu?${params.toString()}`);
}

export function fetchGpuDetails(model) {
  return apiFetch(`/api/products/gpu/${encodeURIComponent(model)}`);
}
